framework/src/main/java/com/github/dbmdz/flusswerk/framework/monitoring/FlusswerkMetrics.java

```diff
@@ -24,6 +32,21 @@ public FlusswerkMetrics(ProcessingProperties properties, MeterRegistry registry)
         .description("Number of worker threads in the system")
         .tags(Tags.of("state", "idle"))
         .register(registry);
+
+    messagesTotal = new EnumMap<>(Status.class);
+    messagesSeconds = new EnumMap<>(Status.class);
+
+    for (Status status : Status.values()) {
+      String statusTag = status.name().toLowerCase(Locale.ROOT);
+      Counter.builder("flusswerk.messages")
+          .tag("status", statusTag)
+          .description("Total number of messages processed since application start")
+          .register(registry);
+      Counter.builder("flusswerk.messages.seconds")
+          .tag("status", statusTag)
+          .description("Total time spent processing messages since application start")
+          .register(registry);
```

Comment:
```suggestion
      messagesTotal.put(
          status,
          Counter.builder("flusswerk.messages")
              .tag("status", statusTag)
              .description("Total number of messages processed since application start")
              .register(registry));
      messagesSeconds.put(
          status,
          Counter.builder("flusswerk.messages.seconds")
              .tag("status", statusTag)
              .description("Total time spent processing messages since application start")
              .register(registry));
```